### PR TITLE
N°6766 - Fix dependent fields not updated due to WizardHelper.UpdateFields() being triggered too early

### PR DESCRIPTION
**Pull request by accognet**
internal

### CODE REVIEW

File: application/wizardhelper.class.inc.php [hunk 1]
```diff
@@ -359,15 +359,38 @@ public function GetJsForUpdateFields()
...
+	public function AddJsForUpdateFields($oPage)
+	{
+		$sWizardHelperJsVar = (!is_null($this->m_aData['m_sWizHelperJsVarName'])) ? utils::Sanitize($this->m_aData['m_sWizHelperJsVarName'], '', utils::ENUM_SANITIZATION_FILTER_PARAMETER) : 'oWizardHelper'.$this->GetFormPrefix();
+		$sWizardHelperJson = $this->ToJSON();
+
+		$oPage->add_script(<<<JS
+{$sWizardHelperJsVar}.m_oData = {$sWizardHelperJson};
+{$sWizardHelperJsVar}.UpdateFields();
+JS
+		);
```
Comment (Hipska) [inline application/wizardhelper.class.inc.php:379]: Duplicate code
```suggestion
		$oPage->add_script($this->GetJsForUpdateFields());
```

**Review by Hipska — CHANGES_REQUESTED** [review]

File: application/wizardhelper.class.inc.php (hunk 1 above)
Comment (accognet) [reply application/wizardhelper.class.inc.php:379]: Yes I know but GetJsForUpdateFields can (must?) be removed with this PR. I need to discuss this change with the team.

Comment (Molkobain) [reply application/wizardhelper.class.inc.php:379]: Original function is now marked as deprecated. The new one allows to feed both regular and on ready JS snippets.

**Review by Molkobain — APPROVED** [review]
Reviewed and accepted

**Pull request merged**